frontend/post/postDetailsDirective.js

```diff
@@ -275,6 +275,33 @@
             return Utils.isLargerThanTheScreen(postDetailsCtrl.post.title) ? 'break' : 'no-break';
         };
 
+        postDetailsCtrl.isSharedEvent = () => {
+            return postDetailsCtrl.post.shared_event; 
+        };
+
+        postDetailsCtrl.isFollowingEvent = () => {
+            const eventFollowers = (postDetailsCtrl.post.shared_event && postDetailsCtrl.post.shared_event.followers) || [];
+            return eventFollowers.includes(postDetailsCtrl.user.key);
+        };
+
+        postDetailsCtrl.followEvent = () => {
+            EventService.addFollower(postDetailsCtrl.post.shared_event.key).then(() => {
+                postDetailsCtrl.post.shared_event.followers.push(postDetailsCtrl.user.key);
+                MessageService.showToast('Você receberá as atualizações desse evento.');
+            }).catch((error) => {
+                console.error(error);
+            });
+        };
+
+        postDetailsCtrl.unFollowEvent = () => {
+            EventService.removeFollower(postDetailsCtrl.post.shared_event.key).then(() => {
+                postDetailsCtrl.post.shared_event.followers = postDetailsCtrl.post.shared_event.followers.filter(follower => follower !== postDetailsCtrl.user.key);
```

Comment:
use event prototype function to remove follower

Reply:
I would have to instantiate the event first, do u think it's worth it?

Reply:
it should have been instantiated already